src/main/java/seedu/address/storage/JsonAdaptedRole.java

```diff
@@ -0,0 +1,128 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.role.Deadline;
+import seedu.address.model.role.Description;
+import seedu.address.model.role.Role;
+import seedu.address.model.role.RoleName;
+import seedu.address.model.role.Status;
+import seedu.address.model.role.Stipend;
+
+public class JsonAdaptedRole {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Company's %s field is missing!";
```

Comment:
Should the missing field message be adapted for a Role?

Reply:
Yup, it definitely should. Thanks for pointing that out!